src/Tasks/Microsoft.NET.Build.Tasks/PrepareForReadyToRunCompilation.cs

```diff
@@ -97,13 +97,51 @@ protected override void ExecuteCore()
 
                 var outputR2RImageRelativePath = file.GetMetadata(MetadataKeys.RelativePath);
                 var outputR2RImage = Path.Combine(OutputPath, outputR2RImageRelativePath);
+                bool crossgen2IsVersion5 = false;
+                if (ReadyToRunUseCrossgen2)
+                {
+                    string isVersion5 = Crossgen2Tool.GetMetadata(MetadataKeys.IsVersion5);
+                    crossgen2IsVersion5 = !string.IsNullOrEmpty(isVersion5) && bool.Parse(isVersion5);
+                }
+
+                string outputPDBImage = null;
+                string outputPDBImageRelativePath = null;
+                string crossgen1CreatePDBCommand = null;
+
+                if (EmitSymbols)
```

Comment:
What happens if `ReadyToRunUseCrossgen2`, `crossgen2IsVersion5`, `Crossgen2Composite`, and `EmitSymbols` are all true? Should we report an error?

Reply:
Good point, error reporting added in 15th commit.